oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/h2/dao/H2MetadataQueryDAO.java

```diff
@@ -146,125 +143,84 @@ public H2MetadataQueryDAO(JDBCHikariCPClient h2Client, int metadataQueryMaxSize)
     }
 
     @Override
-    public Service searchService(NodeType nodeType, String serviceCode) throws IOException {
+    public Service findService(final String serviceId) throws IOException {
         StringBuilder sql = new StringBuilder();
         List<Object> condition = new ArrayList<>(5);
         sql.append("select * from ").append(ServiceTraffic.INDEX_NAME).append(" where ");
-        sql.append(ServiceTraffic.NODE_TYPE).append("=?");
-        condition.add(nodeType.value());
-        sql.append(" and ").append(ServiceTraffic.NAME).append(" = ?");
-        condition.add(serviceCode);
+        sql.append(ServiceTraffic.SERVICE_ID).append(" = ?");
+        condition.add(serviceId);
 
         try (Connection connection = h2Client.getConnection()) {
-            try (ResultSet resultSet = h2Client.executeQuery(
-                connection, sql.toString(), condition.toArray(new Object[0]))) {
-
-                while (resultSet.next()) {
-                    Service service = new Service();
-                    service.setId(resultSet.getString(H2TableInstaller.ID_COLUMN));
-                    service.setName(resultSet.getString(ServiceTraffic.NAME));
-                    service.setGroup(resultSet.getString(ServiceTraffic.GROUP));
-                    return service;
-                }
-            }
+            ResultSet resultSet = h2Client.executeQuery(
+                connection, sql.toString(), condition.toArray(new Object[0]));
+            final List<Service> services = buildServices(resultSet);
+            return services.size() > 0 ? services.get(0) : null;
         } catch (SQLException e) {
             throw new IOException(e);
         }
-
-        return null;
     }
 
     @Override
-    public List<Endpoint> searchEndpoint(String keyword, String serviceId, int limit) throws IOException {
+    public ServiceInstance getInstance(final String instanceId) throws IOException {
         StringBuilder sql = new StringBuilder();
         List<Object> condition = new ArrayList<>(5);
-        sql.append("select * from ").append(EndpointTraffic.INDEX_NAME).append(" where ");
-        sql.append(EndpointTraffic.SERVICE_ID).append("=?");
-        condition.add(serviceId);
-        if (!Strings.isNullOrEmpty(keyword)) {
-            sql.append(" and ").append(EndpointTraffic.NAME).append(" like concat('%',?,'%') ");
-            condition.add(keyword);
-        }
-        sql.append(" limit ").append(limit);
+        sql.append("select * from ").append(InstanceTraffic.INDEX_NAME).append(" where ");
+        sql.append(H2TableInstaller.ID_COLUMN).append(" = ?");
```

Comment:
I don't think we have stored the instance id as column `H2TableInstaller.ID_COLUMN`

Reply:
I guess similar to https://github.com/apache/skywalking/pull/8367#discussion_r776910074.